goaviatrix/edge_neo_device_onbarding.go

```diff
@@ -0,0 +1,110 @@
+package goaviatrix
+
+import (
+	"context"
+)
+
+type EdgeNEODevice struct {
+	Action        string                  `json:"action,omitempty"`
+	CID           string                  `json:"CID,omitempty"`
+	AccountName   string                  `json:"account_name,omitempty"`
+	DeviceName    string                  `json:"device_name,omitempty"`
+	SerialNumber  string                  `json:"serial,omitempty"`
+	HardwareModel string                  `json:"hardware_model,omitempty"`
+	Network       []*EdgeNEODeviceNetwork `json:"network,omitempty"`
+}
+
+type EdgeNEODeviceNetwork struct {
+	InterfaceName string `json:"interface,omitempty"`
+	EnableDhcp    bool   `json:"dhcp,omitempty"`
+	GatewayIp     string `json:"gateway,omitempty"`
+	SubnetCidr    string `json:"subnetCIDR,omitempty"`
+	DnsServerIp   string `json:"dns,omitempty"`
+	ProxyServerIp string `json:"proxy,omitempty"`
+	Tags          string `json:"tags,omitempty"`
+}
+
+//"interface": "eth1",
+//"dhcp": true,
+//"gateway": "",
+//"subnetCIDR": "",
+//"dns": null,
+//"proxy": "",
+//"tags": null
+//
+//"interface": "eth4",
+//"dhcp": true,
+//"gateway": "",
+//"ipv4cidr": "",
+//"dns": null,
+//"proxy": "",
+//"ntpServer": ""
+
+type EdgeNEODeviceResp struct {
+	AccountName   string                  `json:"account_name"`
+	DeviceName    string                  `json:"name"`
+	DeviceId      string                  `json:"deviceId"`
+	SerialNumber  string                  `json:"serial"`
+	HardwareModel string                  `json:"hardwareId"`
+	Network       []*EdgeNEODeviceNetwork `json:"interfaces"`
+}
+
+type EdgeNEODeviceListResp struct {
+	Return  bool                `json:"return"`
+	Results []EdgeNEODeviceResp `json:"results"`
+	Reason  string              `json:"reason"`
+}
+
+func (c *Client) OnboardEdgeNEODevice(ctx context.Context, edgeNEODevice *EdgeNEODevice) error {
```

Comment:
Not sure if this is valuable information, but can also update an onboard device.